internal/aks/client.go

```diff
@@ -0,0 +1,136 @@
+package aks
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2020-11-01/containerservice"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-10-01/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/rancher/aks-operator/internal/utils"
+	aksv1 "github.com/rancher/aks-operator/pkg/apis/aks.cattle.io/v1"
+	wranglerv1 "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+)
+
+type Credentials struct {
+	AuthBaseURL    *string
+	BaseURL        *string
+	SubscriptionID string
+	TenantID       string
+	ClientID       string
+	ClientSecret   string
+}
+
+func NewResourceGroupClient(cred *Credentials) (*resources.GroupsClient, error) {
+	authorizer, err := NewClientAuthorizer(cred)
+	if err != nil {
+		return nil, err
+	}
+
+	if cred.BaseURL == nil {
+		cred.BaseURL = to.StringPtr(azure.PublicCloud.ResourceManagerEndpoint)
+	}
```

Comment:
This is already in `NewClientAuthorizer`. Can be removed.

Reply:
Ok